src/components/PopupWithForm.js

```diff
@@ -2,13 +2,13 @@ import Popup from "./Popup";
 
 export default class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
-    super(popupSelector);
+    super({ popupSelector });
     this._handleFormSubmit = handleFormSubmit;
     this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._saveButton = this._popupForm.querySelector(".modal__button");
   }
 
   _getInputValues() {
-    // const inputs = this._popupForm.querySelectorAll(".modal__input");
     const values = {};
     this._popupForm.querySelectorAll(".modal__input").forEach((input) => {
```

Comment:
**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.

Reply:
Thank you for the comments! I will definitely review them and learn why they're improvements so I can understand how to apply the information in future code 🙏🏼